Geometry/VeryForwardGeometryBuilder/plugins/CTPPSGeometryESModule.cc

```diff
@@ -55,50 +55,80 @@ class CTPPSGeometryESModule : public edm::ESProducer
     CTPPSGeometryESModule( const edm::ParameterSet& );
     ~CTPPSGeometryESModule() override {}
 
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  private:
     std::unique_ptr<DetGeomDesc> produceIdealGD( const IdealGeometryRecord& );
 
+    template<typename ALIGNMENT_REC> struct GDTokens {
+      explicit GDTokens( edm::ESConsumesCollector&& iCC):
+        idealGDToken_{iCC.consumesFrom<DetGeomDesc, IdealGeometryRecord>(edm::ESInputTag())},
+        alignmentToken_{iCC.consumesFrom<CTPPSRPAlignmentCorrectionsData, ALIGNMENT_REC>(edm::ESInputTag())}
+      {}
+      const edm::ESGetToken<DetGeomDesc, IdealGeometryRecord> idealGDToken_;
+      const edm::ESGetToken<CTPPSRPAlignmentCorrectionsData, ALIGNMENT_REC> alignmentToken_;
+    };
+
     std::unique_ptr<DetGeomDesc> produceRealGD( const VeryForwardRealGeometryRecord& );
     std::unique_ptr<CTPPSGeometry> produceRealTG( const VeryForwardRealGeometryRecord& );
 
     std::unique_ptr<DetGeomDesc> produceMisalignedGD( const VeryForwardMisalignedGeometryRecord& );
     std::unique_ptr<CTPPSGeometry> produceMisalignedTG( const VeryForwardMisalignedGeometryRecord& );
 
-  protected:
-    static void applyAlignments( const edm::ESHandle<DetGeomDesc>&, const edm::ESHandle<CTPPSRPAlignmentCorrectionsData>&, DetGeomDesc*& );
+    template<typename REC>
+    std::unique_ptr<DetGeomDesc> produceGD( IdealGeometryRecord const&, const std::optional<REC>&, GDTokens<REC> const&, const char* name);
+    
+
+    static void applyAlignments( const DetGeomDesc&, const CTPPSRPAlignmentCorrectionsData*, DetGeomDesc*& );
     static void buildDetGeomDesc( DDFilteredView* fv, DetGeomDesc* gd );
 
-    unsigned int verbosity_;
-    std::string compactViewTag_;
+    const unsigned int verbosity_;
+    const edm::ESGetToken<DDCompactView, IdealGeometryRecord> compactViewToken_;
+
+    const GDTokens<RPRealAlignmentRecord> gdRealTokens_;
+    const GDTokens<RPMisalignedAlignmentRecord> gdMisTokens_;
+
+    const edm::ESGetToken<DetGeomDesc, VeryForwardRealGeometryRecord> dgdRealToken_;
+    const edm::ESGetToken<DetGeomDesc, VeryForwardMisalignedGeometryRecord> dgdMisToken_;
+
 };
 
 
 //----------------------------------------------------------------------------------------------------
 //----------------------------------------------------------------------------------------------------
 
 CTPPSGeometryESModule::CTPPSGeometryESModule(const edm::ParameterSet& iConfig) :
-  verbosity_( iConfig.getUntrackedParameter<unsigned int>( "verbosity", 1 ) ),
-  compactViewTag_( iConfig.getParameter<std::string>( "compactViewTag" ) )
+  verbosity_( iConfig.getUntrackedParameter<unsigned int>( "verbosity") ),
```

Comment:
```suggestion
  verbosity_{ iConfig.getUntrackedParameter<unsigned int>( "verbosity") },
```

Reply:
Should the use of braces be consistent in this constructor? Probably not important enough to change at this point.

Reply:
I would expect that in the not too distant future we will use clang-tidy to enforce such stylistic consistency.

Reply:
From the discussion we have had it is unclear to which extent the clang-tidy braces are reliable for a regular use